internal/routecapture/networkExtractors.go

```diff
@@ -0,0 +1,120 @@
+package routecapture
+
+import (
+	"context"
+	"net/url"
+	"time"
+
+	webscan "github.com/Method-Security/webscan/generated/go"
+	capture "github.com/Method-Security/webscan/internal/capture"
+	"github.com/go-rod/rod"
+	"github.com/go-rod/rod/lib/proto"
+	"github.com/palantir/witchcraft-go-logging/wlog/svclog/svc1log"
+)
+
+// extractNetworkRoutes fetches network requests, parses them, and populates []WebRoute.
+func extractNetworkRoutes(ctx context.Context, b *capture.BrowserPageCapturer, target string, baseURLsOnly bool) ([]*webscan.WebRoute, []string, []string) {
+	routes := []*webscan.WebRoute{}
+	urls := make(map[string]struct{})
+	errors := []string{}
+	log := svc1log.FromContext(ctx)
+
+	log.Info("Initiating network events capture with browser method", svc1log.SafeParam("target", target))
+	// Ensure the browser is initialized
+	if b.Browser == nil {
+		log.Debug("Initializing browser for network capture")
+		b.InitializeBrowser()
+	}
+
+	// Set up a page with timeout context
+	pageCtx, cancel := context.WithTimeout(ctx, time.Duration(b.TimeoutSeconds)*time.Second)
+	defer cancel()
+
+	var page *rod.Page
+	pageErr := rod.Try(func() {
+		page = b.Browser.MustPage(target).Context(pageCtx)
+	})
+	if pageErr != nil {
+		log.Error("Failed to create page", svc1log.SafeParam("url", target), svc1log.SafeParam("error", pageErr))
+		errors = append(errors, pageErr.Error())
+		return routes, setToListString(urls), errors
+	}
+	log.Debug("Successfully connected to page for network capture")
+
+	// Stabalize DOM
+	page = page.MustWaitDOMStable()
+
+	// Enable network event tracking
+	networkEventErr := proto.NetworkEnable{}.Call(page)
+	if networkEventErr != nil {
+		log.Error("Failed to enable network tracking", svc1log.SafeParam("error", networkEventErr))
+		errors = append(errors, networkEventErr.Error())
+		return routes, setToListString(urls), errors
+	}
+
+	// Capture network requests of type 'fetch' which are typical of API calls
+	networkEvents := []*proto.NetworkRequestWillBeSent{}
+	waitForNetworkEvents := page.EachEvent(func(e *proto.NetworkRequestWillBeSent) {
+		log.Debug("Captured network event", svc1log.SafeParam("url", e.Request.URL), svc1log.SafeParam("type", e.Type))
+		// Only capture fetch requests
+		if e.Type == proto.NetworkResourceTypeFetch {
+			networkEvents = append(networkEvents, e)
+		}
+	})
+
+	// Navigate to the page and wait for the page to load
+	page.MustNavigate(target).MustWaitLoad()
```

Comment:
So prior we wait for the DOM to stabilize, but here we navigate to it...again?

Reply:
i think the dom stabalize thing was redundant